libraries/stumbler/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/scanners/WifiScanner.java

```diff
@@ -153,13 +156,11 @@ synchronized void activatePeriodicScan(final ActiveOrPassiveStumbling stumblingM
         // Ensure that we are constantly scanning for new access points.
         mWifiScanTimer = new Timer();
         mWifiScanTimer.schedule(new TimerTask() {
-            int mPassiveScanCount;
 
             @Override
             public void run() {
                 if (stumblingMode == ActiveOrPassiveStumbling.PASSIVE_STUMBLING &&
-                        mPassiveScanCount++ > AppGlobals.PASSIVE_MODE_MAX_SCANS_PER_GPS) {
```

Comment:
@garvankeeley The way it was before, the scanner will always do `PASSIVE_MODE_MAX_SCANS_PER_GPS` scans and then always `stop()`.
But it can happen that `ScanManager` is calling `start()` on the scanners faster than the scanner can to its scans, so I changed the code to reset the scan count in `start()` instead, to avoid unnecessary stops. And I think that you need `AtomicInteger` for that to be safe. That's why it is now `mPassiveScanCount.incrementAndGet()`. All related changes are  bundled in my second commit.